src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -707,120 +925,94 @@ namespace nbl::ext::imgui
 			}
 
-			// Will project scissor/clipping rectangles into frame-buffer space
-			ImVec2 const clip_off = drawData->DisplayPos;         // (0,0) unless using multi-viewports
-			ImVec2 const clip_scale = drawData->FramebufferScale; // (1,1) unless using retina display which are often (2,2)
-
-			// Render command lists
-			// (Because we merged all buffers into a single one, we maintain our own offset into them)
-			int global_vtx_offset = 0;
-			int global_idx_offset = 0;
-			for (int n = 0; n < drawData->CmdListsCount; n++)
+			const asset::SBufferBinding<const video::IGPUBuffer> binding =
 			{
-				const ImDrawList* cmd_list = drawData->CmdLists[n];
-				for (int cmd_i = 0; cmd_i < cmd_list->CmdBuffer.Size; cmd_i++)
-				{
-					const ImDrawCmd* pcmd = &cmd_list->CmdBuffer[cmd_i];
+				.offset = multiAllocParams.offsets[MDI::EBC_DRAW_INDIRECT_STRUCTURES],
+				.buffer = core::smart_refctd_ptr(mdiBuffer)
+			};
 
-					// Project scissor/clipping rectangles into frame-buffer space
-					ImVec4 clip_rect;
-					clip_rect.x = (pcmd->ClipRect.x - clip_off.x) * clip_scale.x;
-					clip_rect.y = (pcmd->ClipRect.y - clip_off.y) * clip_scale.y;
-					clip_rect.z = (pcmd->ClipRect.z - clip_off.x) * clip_scale.x;
-					clip_rect.w = (pcmd->ClipRect.w - clip_off.y) * clip_scale.y;
+			commandBuffer->drawIndexedIndirect(binding, drawCount, sizeof(VkDrawIndexedIndirectCommand));
 
-					if (clip_rect.x < frameBufferWidth && clip_rect.y < frameBufferHeight && clip_rect.z >= 0.0f && clip_rect.w >= 0.0f)
-					{
-						// Negative offsets are illegal for vkCmdSetScissor
-						if (clip_rect.x < 0.0f)
-							clip_rect.x = 0.0f;
-						if (clip_rect.y < 0.0f)
-							clip_rect.y = 0.0f;
-
-						{// Apply scissor/clipping rectangle
-							VkRect2D scissor{};
-							scissor.offset.x = static_cast<int32_t>(clip_rect.x);
-							scissor.offset.y = static_cast<int32_t>(clip_rect.y);
-							scissor.extent.width = static_cast<uint32_t>(clip_rect.z - clip_rect.x);
-							scissor.extent.height = static_cast<uint32_t>(clip_rect.w - clip_rect.y);
-							commandBuffer->setScissor(0, 1, &scissor);
-						}
-
-						// Draw
-						commandBuffer->drawIndexed(
-							pcmd->ElemCount,
-							1,
-							pcmd->IdxOffset + global_idx_offset,
-							pcmd->VtxOffset + global_vtx_offset,
-							0
-						);
-					}
-				}
-				global_idx_offset += cmd_list->IdxBuffer.Size;
-				global_vtx_offset += cmd_list->VtxBuffer.Size;
+			scratchSemaphoreCounters.newie = info.scratchSemaphore.value;
+
+			if (scratchSemaphoreCounters.newie != scratchSemaphoreCounters.oldie)
+			{
+				const auto overflows = scratchSemaphoreCounters.newie - scratchSemaphoreCounters.oldie;
+				logger->log("%d overflows when rendering UI!\n", nbl::system::ILogger::ELL_PERFORMANCE, overflows);
+
+				// TODO: handle them?
+				exit(0x45);
 			}
```

Comment:
the way you wrote your function doesn't call `overflowSubmit` but when it will, you don't really need to do anything about the overflows

You can just count how many times you called `overflowSubmit` yourself

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/707#discussion_r1763267517